docs/docs/installation.md

```diff
@@ -48,6 +48,38 @@ However, to setup for development we recommend using the makefile setup which wi
 make init-dev # installs pre-commit as a hook
 ```
 
+### 📦 Upgrading Packages
+
+To upgrade dependencies in your development environment:
+
+```bash
+# Upgrade all packages to their latest compatible versions
+uv sync --upgrade
```

Comment:
Would it be worth adding a recommendation to use `--exclude-newer` to these commands to avoid upgrading to brand new packages?